### PR TITLE
Refactor database configuration and make the main API typed

### PR DESCRIPTION
**Pull request by oranenj**
Refactors DB configuration to use a single template.

Changes PostgreSQL defaults a bit, but I'm pretty sure the default was ineffective anyway, so this is half of an API break...

### CODE REVIEW

**Comment by oranenj** [comment]
Squashed and reworded a bit

**Review by bastelfreak — APPROVED** [review]

**Comment by oranenj** [comment]
Needs another review, the changes are a bit more drastic now.

Basically, set defaults for ~everything to undef and omit them from dbconfig.xml unless specified by the user.

Also removes poolsize, I don't think it does anything, since it's not documented by Atlassian.

File: manifests/init.pp [hunk 1]
```diff
@@ -99,50 +97,50 @@
...
   $deploy_module                                                    = 'archive',
-  $proxy_server                                                     = undef,
+  Optional[String] $proxy_server                                    = undef,
   Optional[Enum['none','http','https','ftp']] $proxy_type           = undef,
   # Manage service
-  $service_manage                                                   = true,
-  $service_ensure                                                   = running,
-  $service_enable                                                   = true,
+  Boolean $service_manage                                           = true,
+  Enum['running', 'stopped'] $service_ensure                        = 'running',
```
Comment (ekohl) [inline manifests/init.pp:108]:
```suggestion
  Stdlib::Ensure::Service $service_ensure                           = 'running',
```

**Review by ekohl — APPROVED** [review]

**Comment by oranenj** [comment]
I've also tested this on the Jira instance I have available and it doesn't break anything obvious, so I'll squash-merge.

**Pull request merged**